From 53e58ffcd6b899fce3e51a28ec92c12e28850a97 Mon Sep 17 00:00:00 2001
From: Christian Kampka <christian@kampka.net>
Date: Tue, 2 Mar 2021 14:59:14 +0100
Subject: [PATCH] Enable config dir setting via env variable

---
 commands/root.go | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/commands/root.go b/commands/root.go
index d6d45c4..f015964 100644
--- a/commands/root.go
+++ b/commands/root.go
@@ -56,8 +56,14 @@ func init() {
 	// if it exists.
 	tokenFromEnv := os.Getenv("GSCTL_AUTH_TOKEN")
 
+	defaultConfigDir := config.DefaultConfigDirPath
+	configDirFromEnv := os.Getenv("GSCTL_CONFIG_DIR")
+	if (configDirFromEnv != "") {
+		defaultConfigDir = configDirFromEnv
+	}
+
 	RootCommand.PersistentFlags().StringVarP(&flags.Token, "auth-token", "", tokenFromEnv, "Authorization token to use")
-	RootCommand.PersistentFlags().StringVarP(&flags.ConfigDirPath, "config-dir", "", config.DefaultConfigDirPath, "Configuration directory path to use")
+	RootCommand.PersistentFlags().StringVarP(&flags.ConfigDirPath, "config-dir", "", defaultConfigDir, "Configuration directory path to use")
 	RootCommand.PersistentFlags().BoolVarP(&flags.Verbose, "verbose", "v", false, "Print more information")
 	RootCommand.PersistentFlags().BoolVarP(&flags.SilenceHTTPEndpointWarning, "silence-http-endpoint-warning", "", false, "Dont't print warnings when deliberately using an insecure HTTP endpoint")
 	RootCommand.Flags().Bool("version", false, version.Command.Short)
-- 
2.29.2

